implementations/rust/ockam/ockam_api/src/kafka/portal_worker.rs

```diff
@@ -239,17 +258,19 @@ impl KafkaPortalWorker {
     /// used for requests.
     pub(crate) async fn start(
         context: &mut Context,
+        secure_channel_controller: Arc<dyn KafkaSecureChannelController>,
+        uuid_to_name: TopicUuidMap,
         inlet_map: KafkaInletMap,
     ) -> ockam_core::Result<Address> {
-        let shared_protocol_state = ProtocolState::new();
+        let shared_protocol_state = ProtocolState::new(secure_channel_controller, uuid_to_name);
```

Comment:
Would it make sense to put the `inlet_map` inside the `ProtocolState` in order to avoid having to pass it to `intercept_response`?

Reply:
The ProtocolState is shared across workers, while the inlet_map is global across every kafka service (granted normally you should have 1).
Then the TopicUuidMap came in within the ProtocolState breaking a bit that separation, yep maybe a review of states is due.